// etag/@mastra/core/dist/signals/index.cjs
'use strict';

var chunkK5JBD726_cjs = require('../chunk-K5JBD726.cjs');

// src/signals/task-signal-provider.ts
var TaskSignalProvider = class extends chunkK5JBD726_cjs.SignalProvider {
  id = "task-signals";
  #processor = new chunkK5JBD726_cjs.TaskStateProcessor();
  getInputProcessors() {
    return [this.#processor];
  }
  getTools() {
    return {
      task_write: chunkK5JBD726_cjs.taskWriteTool,
      task_update: chunkK5JBD726_cjs.taskUpdateTool,
      task_complete: chunkK5JBD726_cjs.taskCompleteTool,
      task_check: chunkK5JBD726_cjs.taskCheckTool
    };
  }
};

Object.defineProperty(exports, "GoalSignalProvider", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.GoalSignalProvider; }
});
Object.defineProperty(exports, "SignalProvider", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.SignalProvider; }
});
Object.defineProperty(exports, "WebhookSignalProvider", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.WebhookSignalProvider; }
});
Object.defineProperty(exports, "assignTaskIds", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.assignTaskIds; }
});
Object.defineProperty(exports, "isSignalProvider", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.isSignalProvider; }
});
exports.TaskSignalProvider = TaskSignalProvider;
