// etag/@mastra/core/dist/tools/index.cjs
'use strict';

var chunkK5JBD726_cjs = require('../chunk-K5JBD726.cjs');
var chunk3XBBTP2N_cjs = require('../chunk-3XBBTP2N.cjs');
var chunkHISQ2TWD_cjs = require('../chunk-HISQ2TWD.cjs');
var chunkKPH2IJW4_cjs = require('../chunk-KPH2IJW4.cjs');



Object.defineProperty(exports, "DEFAULT_GOAL_JUDGE_PROMPT", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.DEFAULT_GOAL_JUDGE_PROMPT; }
});
Object.defineProperty(exports, "DEFAULT_GOAL_MAX_RUNS", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.DEFAULT_GOAL_MAX_RUNS; }
});
Object.defineProperty(exports, "FRAME_PREFIX", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.FRAME_PREFIX; }
});
Object.defineProperty(exports, "GOAL_STATE_ID", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.GOAL_STATE_ID; }
});
Object.defineProperty(exports, "GOAL_STATE_TYPE", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.GOAL_STATE_TYPE; }
});
Object.defineProperty(exports, "GoalStateProcessor", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.GoalStateProcessor; }
});
Object.defineProperty(exports, "StdioCodeModeTransport", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.StdioCodeModeTransport; }
});
Object.defineProperty(exports, "TASKS_REQUEST_CONTEXT_KEY", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.TASKS_REQUEST_CONTEXT_KEY; }
});
Object.defineProperty(exports, "TASKS_STATE_ID", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.TASKS_STATE_ID; }
});
Object.defineProperty(exports, "TASK_STATE_TYPE", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.TASK_STATE_TYPE; }
});
Object.defineProperty(exports, "TaskStateProcessor", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.TaskStateProcessor; }
});
Object.defineProperty(exports, "askUserTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.askUserTool; }
});
Object.defineProperty(exports, "assignTaskIds", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.assignTaskIds; }
});
Object.defineProperty(exports, "buildProgramModule", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.buildProgramModule; }
});
Object.defineProperty(exports, "buildRunner", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.buildRunner; }
});
Object.defineProperty(exports, "clearObjective", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.clearObjective; }
});
Object.defineProperty(exports, "createCodeMode", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createCodeMode; }
});
Object.defineProperty(exports, "createCodeModeInstructions", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createCodeModeInstructions; }
});
Object.defineProperty(exports, "createCodeModeTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createCodeModeTool; }
});
Object.defineProperty(exports, "demoteExtraInProgress", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.demoteExtraInProgress; }
});
Object.defineProperty(exports, "formatQuestionAnswer", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.formatQuestionAnswer; }
});
Object.defineProperty(exports, "formatTaskListResult", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.formatTaskListResult; }
});
Object.defineProperty(exports, "generateStubs", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.generateStubs; }
});
Object.defineProperty(exports, "getObjectiveFromRequestContext", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.getObjectiveFromRequestContext; }
});
Object.defineProperty(exports, "getTasksFromRequestContext", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.getTasksFromRequestContext; }
});
Object.defineProperty(exports, "hasMultipleInProgress", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.hasMultipleInProgress; }
});
Object.defineProperty(exports, "jsonSchemaToTsString", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.jsonSchemaToTsString; }
});
Object.defineProperty(exports, "readObjective", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.readObjective; }
});
Object.defineProperty(exports, "resolveEffectiveGoalSettings", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.resolveEffectiveGoalSettings; }
});
Object.defineProperty(exports, "resolveGoalStore", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.resolveGoalStore; }
});
Object.defineProperty(exports, "submitPlanTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.submitPlanTool; }
});
Object.defineProperty(exports, "summarizeTaskCheck", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.summarizeTaskCheck; }
});
Object.defineProperty(exports, "taskCheckTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.taskCheckTool; }
});
Object.defineProperty(exports, "taskCompleteTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.taskCompleteTool; }
});
Object.defineProperty(exports, "taskUpdateTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.taskUpdateTool; }
});
Object.defineProperty(exports, "taskWriteTool", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.taskWriteTool; }
});
Object.defineProperty(exports, "writeObjective", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.writeObjective; }
});
Object.defineProperty(exports, "ToolStream", {
  enumerable: true,
  get: function () { return chunk3XBBTP2N_cjs.ToolStream; }
});
Object.defineProperty(exports, "noopObserve", {
  enumerable: true,
  get: function () { return chunk3XBBTP2N_cjs.noopObserve; }
});
Object.defineProperty(exports, "MASTRA_TOOL_MARKER", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.MASTRA_TOOL_MARKER; }
});
Object.defineProperty(exports, "Tool", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.Tool; }
});
Object.defineProperty(exports, "createTool", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.createTool; }
});
Object.defineProperty(exports, "isProviderDefinedTool", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.isProviderDefinedTool; }
});
Object.defineProperty(exports, "isProviderTool", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.isProviderTool; }
});
Object.defineProperty(exports, "isValidationError", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.isValidationError; }
});
Object.defineProperty(exports, "isVercelTool", {
  enumerable: true,
  get: function () { return chunkHISQ2TWD_cjs.isVercelTool; }
});
Object.defineProperty(exports, "getTransformedToolPayload", {
  enumerable: true,
  get: function () { return chunkKPH2IJW4_cjs.getTransformedToolPayload; }
});
Object.defineProperty(exports, "hasTransformedToolPayload", {
  enumerable: true,
  get: function () { return chunkKPH2IJW4_cjs.hasTransformedToolPayload; }
});
