// etag/@mastra/core/dist/llm/index.cjs
'use strict';

var chunkYK5LWDCV_cjs = require('../chunk-YK5LWDCV.cjs');
var chunkZ4KN2GTH_cjs = require('../chunk-Z4KN2GTH.cjs');
var chunkAWV2IL46_cjs = require('../chunk-AWV2IL46.cjs');
var chunkYYUDVZJC_cjs = require('../chunk-YYUDVZJC.cjs');
var chunkULG4VSW4_cjs = require('../chunk-ULG4VSW4.cjs');
var chunkYIPJLB6J_cjs = require('../chunk-YIPJLB6J.cjs');



Object.defineProperty(exports, "EMBEDDING_MODELS", {
  enumerable: true,
  get: function () { return chunkYK5LWDCV_cjs.EMBEDDING_MODELS; }
});
Object.defineProperty(exports, "ModelRouterEmbeddingModel", {
  enumerable: true,
  get: function () { return chunkYK5LWDCV_cjs.ModelRouterEmbeddingModel; }
});
Object.defineProperty(exports, "ModelRouterLanguageModel", {
  enumerable: true,
  get: function () { return chunkYK5LWDCV_cjs.ModelRouterLanguageModel; }
});
Object.defineProperty(exports, "resolveModelAuth", {
  enumerable: true,
  get: function () { return chunkYK5LWDCV_cjs.resolveModelAuth; }
});
Object.defineProperty(exports, "resolveModelConfig", {
  enumerable: true,
  get: function () { return chunkYK5LWDCV_cjs.resolveModelConfig; }
});
Object.defineProperty(exports, "AzureOpenAIGateway", {
  enumerable: true,
  get: function () { return chunkZ4KN2GTH_cjs.AzureOpenAIGateway; }
});
Object.defineProperty(exports, "defaultGateways", {
  enumerable: true,
  get: function () { return chunkZ4KN2GTH_cjs.defaultGateways; }
});
Object.defineProperty(exports, "GatewayRegistry", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.GatewayRegistry; }
});
Object.defineProperty(exports, "MastraGateway", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.MastraGateway; }
});
Object.defineProperty(exports, "PROVIDER_REGISTRY", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.PROVIDER_REGISTRY; }
});
Object.defineProperty(exports, "getProviderConfig", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.getProviderConfig; }
});
Object.defineProperty(exports, "modelSupportsAttachments", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.modelSupportsAttachments; }
});
Object.defineProperty(exports, "parseModelString", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.parseModelString; }
});
Object.defineProperty(exports, "ModelsDevGateway", {
  enumerable: true,
  get: function () { return chunkYYUDVZJC_cjs.ModelsDevGateway; }
});
Object.defineProperty(exports, "NetlifyGateway", {
  enumerable: true,
  get: function () { return chunkULG4VSW4_cjs.NetlifyGateway; }
});
Object.defineProperty(exports, "GATEWAY_AUTH_HEADER", {
  enumerable: true,
  get: function () { return chunkYIPJLB6J_cjs.GATEWAY_AUTH_HEADER; }
});
Object.defineProperty(exports, "MastraModelGateway", {
  enumerable: true,
  get: function () { return chunkYIPJLB6J_cjs.MastraModelGateway; }
});
