// etag/@mastra/core/dist/background-tasks/index.cjs
'use strict';

var chunkERDO25O4_cjs = require('../chunk-ERDO25O4.cjs');



Object.defineProperty(exports, "BACKGROUND_TASK_WORKFLOW_ID", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.BACKGROUND_TASK_WORKFLOW_ID; }
});
Object.defineProperty(exports, "BackgroundTaskManager", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.BackgroundTaskManager; }
});
Object.defineProperty(exports, "backgroundOverrideJsonSchema", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.backgroundOverrideJsonSchema; }
});
Object.defineProperty(exports, "backgroundOverrideZodSchema", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.backgroundOverrideZodSchema; }
});
Object.defineProperty(exports, "createBackgroundTask", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.createBackgroundTask; }
});
Object.defineProperty(exports, "generateBackgroundTaskSystemPrompt", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.generateBackgroundTaskSystemPrompt; }
});
Object.defineProperty(exports, "resolveBackgroundConfig", {
  enumerable: true,
  get: function () { return chunkERDO25O4_cjs.resolveBackgroundConfig; }
});
