// etag/@mastra/core/dist/workflows/index.cjs
'use strict';

var chunkK5JBD726_cjs = require('../chunk-K5JBD726.cjs');

// src/workflows/state-reader.ts
var getStep = (state, stepId) => state.steps?.[stepId];
var getFirstStepResult = (step) => {
  return Array.isArray(step) ? step.find((result) => result?.status === "suspended") ?? step[0] : step;
};
var getNestedSuspendPath = (step) => {
  const path = getFirstStepResult(step)?.suspendPayload?.__workflow_meta?.path;
  return Array.isArray(path) ? path.filter((part) => typeof part === "string") : [];
};
function getWorkflowStepOutput(state, stepId) {
  const step = getStep(state, stepId);
  return Array.isArray(step) ? step.map((result) => result?.output) : step?.output;
}
function getWorkflowStepPayload(state, stepId) {
  const step = getStep(state, stepId);
  return Array.isArray(step) ? step.map((result) => result?.payload) : step?.payload;
}
function getWorkflowResumeLabel(state, label) {
  const resumeLabel = state.resumeLabels?.[label];
  return resumeLabel ? { ...resumeLabel } : void 0;
}
function getWorkflowResumeLabels(state) {
  return Object.entries(state.resumeLabels ?? {}).reduce(
    (labels, [label, value]) => {
      labels[label] = { ...value };
      return labels;
    },
    {}
  );
}
function getWorkflowSuspendedSteps(state) {
  return Object.entries(state.suspendedPaths ?? {}).map(([stepId, executionPath]) => {
    const step = getStep(state, stepId);
    const firstStepResult = getFirstStepResult(step);
    const nestedPath = getNestedSuspendPath(step);
    const path = nestedPath.length > 0 ? nestedPath[0] === stepId ? nestedPath : [stepId, ...nestedPath] : [stepId];
    const resumeLabels = Object.entries(state.resumeLabels ?? {}).reduce(
      (labels, [label, value]) => {
        if (value.stepId === stepId) {
          labels[label] = { ...value };
        }
        return labels;
      },
      {}
    );
    return {
      stepId,
      path,
      executionPath,
      step,
      payload: Array.isArray(step) ? step.map((result) => result?.payload) : step?.payload,
      suspendPayload: firstStepResult?.suspendPayload,
      suspendOutput: firstStepResult?.suspendOutput,
      resumeLabels
    };
  });
}
function getWorkflowSuspendedStep(state) {
  return getWorkflowSuspendedSteps(state)[0];
}
function createWorkflowStateReader(state) {
  return {
    getStatus: () => state.status,
    getResult: () => state.result,
    getError: () => state.error,
    getStepOutput: (stepId) => getWorkflowStepOutput(state, stepId),
    getStepPayload: (stepId) => getWorkflowStepPayload(state, stepId),
    getSuspendedStep: () => getWorkflowSuspendedStep(state),
    getSuspendedSteps: () => getWorkflowSuspendedSteps(state),
    getResumeLabel: (label) => getWorkflowResumeLabel(state, label),
    getResumeLabels: () => getWorkflowResumeLabels(state)
  };
}

Object.defineProperty(exports, "DefaultExecutionEngine", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.DefaultExecutionEngine; }
});
Object.defineProperty(exports, "ExecutionEngine", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.ExecutionEngine; }
});
Object.defineProperty(exports, "Run", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.Run; }
});
Object.defineProperty(exports, "Workflow", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.Workflow; }
});
Object.defineProperty(exports, "WorkflowScheduler", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.WorkflowScheduler; }
});
Object.defineProperty(exports, "cleanStepResult", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.cleanStepResult; }
});
Object.defineProperty(exports, "cloneStep", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.cloneStep; }
});
Object.defineProperty(exports, "cloneWorkflow", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.cloneWorkflow; }
});
Object.defineProperty(exports, "computeNextFireAt", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.computeNextFireAt; }
});
Object.defineProperty(exports, "createDeprecationProxy", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createDeprecationProxy; }
});
Object.defineProperty(exports, "createEventedWorkflow", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createEventedWorkflow; }
});
Object.defineProperty(exports, "createRestartExecutionParams", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createRestartExecutionParams; }
});
Object.defineProperty(exports, "createStep", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createStep; }
});
Object.defineProperty(exports, "createTimeTravelExecutionParams", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createTimeTravelExecutionParams; }
});
Object.defineProperty(exports, "createWorkflow", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.createWorkflow; }
});
Object.defineProperty(exports, "getResumeLabelsByStepId", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.getResumeLabelsByStepId; }
});
Object.defineProperty(exports, "getStepIds", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.getStepIds; }
});
Object.defineProperty(exports, "getStepResult", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.getStepResult; }
});
Object.defineProperty(exports, "hydrateSerializedStepErrors", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.hydrateSerializedStepErrors; }
});
Object.defineProperty(exports, "isProcessor", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.isProcessor; }
});
Object.defineProperty(exports, "mapVariable", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.mapVariable; }
});
Object.defineProperty(exports, "runCountDeprecationMessage", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.runCountDeprecationMessage; }
});
Object.defineProperty(exports, "validateCron", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateCron; }
});
Object.defineProperty(exports, "validateStepInput", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateStepInput; }
});
Object.defineProperty(exports, "validateStepRequestContext", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateStepRequestContext; }
});
Object.defineProperty(exports, "validateStepResumeData", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateStepResumeData; }
});
Object.defineProperty(exports, "validateStepStateData", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateStepStateData; }
});
Object.defineProperty(exports, "validateStepSuspendData", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.validateStepSuspendData; }
});
Object.defineProperty(exports, "waitForSuspendedSnapshot", {
  enumerable: true,
  get: function () { return chunkK5JBD726_cjs.waitForSuspendedSnapshot; }
});
exports.createWorkflowStateReader = createWorkflowStateReader;
exports.getWorkflowResumeLabel = getWorkflowResumeLabel;
exports.getWorkflowResumeLabels = getWorkflowResumeLabels;
exports.getWorkflowStepOutput = getWorkflowStepOutput;
exports.getWorkflowStepPayload = getWorkflowStepPayload;
exports.getWorkflowSuspendedStep = getWorkflowSuspendedStep;
exports.getWorkflowSuspendedSteps = getWorkflowSuspendedSteps;
