// etag/@mastra/core/dist/provider-registry-B2CJZB7I.cjs
'use strict';

var chunkAWV2IL46_cjs = require('./chunk-AWV2IL46.cjs');



Object.defineProperty(exports, "GatewayRegistry", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.GatewayRegistry; }
});
Object.defineProperty(exports, "PROVIDER_MODELS", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.PROVIDER_MODELS; }
});
Object.defineProperty(exports, "PROVIDER_REGISTRY", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.PROVIDER_REGISTRY; }
});
Object.defineProperty(exports, "getProviderConfig", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.getProviderConfig; }
});
Object.defineProperty(exports, "getRegisteredProviders", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.getRegisteredProviders; }
});
Object.defineProperty(exports, "isOfflineMode", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.isOfflineMode; }
});
Object.defineProperty(exports, "isProviderRegistered", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.isProviderRegistered; }
});
Object.defineProperty(exports, "isValidModelId", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.isValidModelId; }
});
Object.defineProperty(exports, "modelSupportsAttachments", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.modelSupportsAttachments; }
});
Object.defineProperty(exports, "parseModelString", {
  enumerable: true,
  get: function () { return chunkAWV2IL46_cjs.parseModelString; }
});
